File: leetcode2/2medium/김대원/Q209.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int answer = Integer.MAX_VALUE;
        int p1 = 0;
        long sum = 0;
        for (int p2 = 0; p2 < nums.length; p2++) {
            sum += nums[p2];

            while (sum >= target) {
                answer = Math.min(answer, p2 - p1 + 1);
                sum -= nums[p1++];
            }
        }

        return answer == Integer.MAX_VALUE ? 0 : answer;
    }
}

File: leetcode2/2medium/강병호/Q209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        nums.sort(reverse=True)

        print(f"nums : {nums}")

        if sum(nums) < target:
            return 0
        
        sum_res = 0
        cnt = 0
        for num in nums:
            sum_res += num
            cnt += 1

            if sum_res >= target:
                print(f"sumres : {sum_res} // cnt : {cnt}")
                return cnt


            